### PR TITLE
Fix k8s.io/sample-apiserver@v1.17.0 that fails the discovery check

### LINKED ISSUE #86155: k8s.io/sample-apiserver@v1.17.0 fails the discovery check
**What happened**:
I follow https://github.com/kubernetes/sample-apiserver/blob/kubernetes-1.17.0/docs/minikube-walkthrough.md to deploy the sample-apiserver and the sample-apiserver server is not available.

**What you expected to happen**:
The sample-apiserver to be available.

**How to reproduce it (as minimally and precisely as possible)**:
1. Follow https://github.com/kubernetes/sample-apiserver/blob/kubernetes-1.17.0/docs/minikube-walkthrough.md
2. Ensure that the discovery check fails
```
$ k api-resources
[...]
csidrivers                                     storage.k8s.io                 false        CSIDriver
csinodes                                       storage.k8s.io                 false        CSINode
storageclasses                    sc           storage.k8s.io                 false        StorageClass
volumeattachments                              storage.k8s.io                 false        VolumeAttachment
error: unable to retrieve the complete list of server APIs: wardle.example.com/v1alpha1: the server is currently unable to handle the request
```

```
$ k logs wardle-server-d66784497-44r8m -n wardle wardle-server
I1211 10:42:28.131791       1 secure_serving.go:178] Serving securely on [::]:443
I1211 10:42:28.132106       1 tlsconfig.go:219] Starting DynamicServingCertificateController
I1211 10:42:28.652093       1 client.go:361] parsed scheme: "endpoint"
I1211 10:42:28.652542       1 endpoint.go:68] ccResolverWrapper: sending new addresses to cc: [{http://localhost:2379 0  <nil>}]
E1211 10:42:29.135510       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:29.144829       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:30.137471       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:30.147896       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:31.139719       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:31.150381       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:32.142401       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:32.152783       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:33.145192       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:33.155008       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:34.146920       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:34.157971       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:35.149089       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:35.159908       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:36.151307       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:36.166394       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:37.153859       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:37.169534       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:38.156138       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:38.171714       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:39.158359       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:39.173455       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:40.161547       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
E1211 10:42:40.174923       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191121015835-571c0ef67034/tools/cache/reflector.go:108: Failed to list *v1.ConfigMap: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:wardle:apiserver" cannot list resource "configmaps" in API group "" in the namespace "kube-system"
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ k version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.1", GitCommit:"b7394102d6ef778017f2ca4046abbaa23b88c290", GitTreeState:"clean", BuildDate:"2019-04-08T17:11:31Z", GoVersion:"go1.12.1", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:13:49Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

- Cloud provider or hardware configuration: minikube
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #86155

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
An issue preventing `k8s.io/sample-apiserver` to handle request is now fixed.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
cc @nikhita @liggitt

File: staging/src/k8s.io/sample-apiserver/artifacts/example/rbac.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ metadata:
   name: aggregated-apiserver-clusterrole
 rules:
 - apiGroups: [""]
-  resources: ["namespaces"]
+  resources: ["namespaces", "configmaps"]
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/rbac.yaml]: global access to configmaps is not desired. the rolebinding to the `extension-apiserver-authentication-reader` role should cover the read access required

**Comment by liggitt** [comment]
The `extension-apiserver-authentication-reader` role (which the example already grants to the wardle apiserver service account) contains the required permissions in 1.17.0+

The reported issue occurred when running against a 1.16.0 Kubernetes API server.

/close

**Comment by ialidzhikov** [comment]
Yep, with 1.17.0 it works.

**Pull request closed without merging**